CodeGen/Generators/NanoFrameworkGenerator.cs

```diff
@@ -126,85 +122,183 @@ public static void Generate(string rootDir, Quantity[] quantities, bool updateNa
             Log.Information("");
         }
 
-        private static void SetDependencyVersions(FindPackageByIdResource resource, SourceCacheContext cache, ILogger logger,
-            CancellationToken cancellationToken, bool updateNanoFrameworkDependencies, string outputDir)
+        /// <summary>
+        /// Updates existing nanoFramework projects and nuspecs with the latest versions.
+        /// </summary>
+        /// <param name="rootDir">The root directory</param>
+        /// <param name="quantities">The quantities to update nuspecs</param>
+        public static void UpdateNanoFrameworkDependencies(
+            string rootDir,
+            Quantity[] quantities)
         {
-            if (updateNanoFrameworkDependencies)
+            // working path
+            string path = Path.Combine(rootDir, "UnitsNet.NanoFramework\\GeneratedCode");
+
+            Log.Information("");
+            Log.Information("Updating .NET nanoFramework references using nuget CLI");
+
+            // run nuget CLI
+            var nugetCLI = new System.Diagnostics.Process
             {
-                logger.LogInformation("Updating nanoFramework dependencies.");
-
-                // mscorlib
-                IEnumerable<NuGetVersion> packageVersions = resource.GetAllVersionsAsync(
-                    "nanoFramework.CoreLibrary",
-                    cache,
-                    logger,
-                    cancellationToken).Result;
-
-                // get NuGet package Version for mscorlib
-                // grab latest available (doesn't matter if it's preview or stable)
-                NuGetVersion mscorlibVersion = packageVersions.OrderByDescending(v => v).First();
-                MscorlibVersion = mscorlibVersion.Version.ToString();
-                MscorlibNuGetVersion = mscorlibVersion.ToNormalizedString();
-
-                // System.Math
-                packageVersions = resource.GetAllVersionsAsync(
-                    "nanoFramework.System.Math",
-                    cache,
-                    logger,
-                    cancellationToken).Result;
-
-                // grab latest available (doesn't matter if it's preview or stable)
-                // making an assumption here that the available version is referencing the correct mscolib
-                var mathVersion = packageVersions.OrderByDescending(v => v).First();
-                MathVersion = mathVersion.Version.ToString();
-                MathNuGetVersion = mathVersion.ToNormalizedString();
-
-                logger.LogInformation($"Referencing nanoFramework.CoreLibrary {MscorlibNuGetVersion}");
-                logger.LogInformation($"Referencing nanoFramework.System.Math {MathNuGetVersion}");
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "nuget.exe",
```

Comment:
I don't have nuget.exe in my PATH on Windows so this fails to run for me.
There is one in `Tools\NuGet.exe` we could use though. If it's too old, just update it.

![image](https://user-images.githubusercontent.com/787816/129488617-69718c29-5594-46c3-9a0c-fd887c30e8f4.png)

Reply:
Pushed a commit with updated nuget.exe and using this from codegen.

Reply:
Actually, I can't seem to push any changes to your fork for some reason. Have you locked it?
I pushed the commits to a branch on my repo.
https://github.com/angularsen/UnitsNet/commits/work-962

Reply:
Just pulled your changes.